src/Router.tsx

```diff
@@ -17,7 +17,7 @@ export default function Router(): ReactElement {
         <NavigationBar />
         <Switch>
           <Route path="/account">{authUser ? <MyAccount /> : <Login />}</Route>
-          <Route path="/promotion">{authUser ? <MyPromotions /> : <Login />}</Route>
+          <Route path="/mypromotions">{authUser ? <MyPromotions /> : <Login />}</Route>
           <Route path="/promotion/upload">{authUser ? <UploadPromotion /> : <Login />}</Route>
```

Comment:
For some reason, if you click promotion to promotion/upload or vice versa, it wouldn't change. I''ve separated the routes since they are actually entirely differente